DM-39097: Start a ci_summit repo

hsinfang opened the pull request.

hsinfang committed:
Set up a ci_builder-based package modeling ci_imsim

hsinfang committed:
Build a butler repo using testdata_ci_summit

hsinfang committed:
Use test utility from daf.butler.tests

hsinfang committed:
Copy tests directly from summit_utils

They cannot be run yet.

Skip test_utils.py as it does not use a butler repo.

Skip test_butlerUtils.py  as it currently targets a specific
butler repo and its contents.

hsinfang committed:
Reformat with black and fix for max-doc-length flake8 complaint

hsinfang committed:
Migrate tests to use the test repo in ci_summit

Instead of relying on the butler repos at specific data
facilities. Move them to use the small test repo instead.

For now, try to stay the same as how they are in summit_utils
as much as possible.

czwa commented on README.rst:
```diff
@@ -3,5 +3,29 @@ ci_summit
+
+#. Clone this package, `ci_builder <https://github.com/lsst-dm/ci_builder>`_, and
+`testdata_ci_summit <https://github.com/lsst-dm/testdata_ci_summit>`_.
+#. ``setup -r ci_builder``
```
This package needs to have `scons` run on it, so I think that needs to be added here as well.

hsinfang replied:
It doesn't need `scons`, and that's the cool thing about `ci_builder`!   `ci_imsim` runs that way too and I use it as the model for this repo. 

Trying to find docs and found [RFC-774](https://jira.lsstcorp.org/browse/RFC-774)

Or do you mean a basic scons file should be added?

czwa replied:
I meant that `ci_builder` needs to have `scons` run on it, since the end user will have a fresh clone of it.  Most people on the project would probably know to do so anyway, but it's always good if the documentation is explicit about things like that.

hsinfang replied:
Got it!  Thank you for clarifying!

czwa commented on README.rst:
```diff
@@ -3,5 +3,29 @@ ci_summit
+#. From the root of this package directory run ``bin/rewrite.sh`` to rewrite python shebang lines.
+#. Run ``bin/ci_summit_run.py`` (see available options with ``--help``).
+
+To cleanup after a run, use either ``bin/ci_summit_run.py --clean`` or ``rm -rf DATA/``.
```
`--clean` doesn't show up as an option when looking at the `--help` options.  I'm unsure if this is a `ci_summit` issue or a `ci_builder` one.

hsinfang replied:
It was a bug in ci_imsim [DM-39176](https://jira.lsstcorp.org/browse/DM-39176) and got fixed, but I started this before that and missed to fix it.   Thanks so much for noticing it!

czwa commented on bin.src/ci_summit_run.py:
```diff
@@ -0,0 +1,54 @@
+    description="Run ci_summit on testdata_ci_summit data",
+)
+args, args_left = parser.parse_known_args()
+sys.argv = sys.argv[:1] + args_left
```
If I'm thinking about this correctly, this argument parser grabs `--help`, displays its help message, which prevents it from being an argument that `ciRunner` checks.  Should this also have `prog="ci_summit_run.py"`?

czwa commented on tests/test_imageExaminer.py:
```diff
@@ -0,0 +1,89 @@
+            doForceCoM=False,
+            savePlots=self.outputFilename,
+        )
+        imExam.plot()
```
I think this test needs the `mpl.use("Agg")` as found in `test_nightReport.py`, as I had a window pop up displaying this figure.

czwa reviewed (approved):
Minor comments, mostly me working out how the system works.

hsinfang committed:
Use matplotlib non-interactive backend

Otherwise a window would pop up displaying this figure.

hsinfang committed:
Make the butler repo writeable

Otherwise registering new collections can fail with:
  lsst.daf.butler.registry.interfaces._database.ReadOnlyDatabaseError:
  sync needs to insert, but database is read-only.
